### ISSUE TITLE
[BUG][2.9] Rancher Agent image manifest list missing `os.version` for windows images

### ISSUE THREAD

**Issue by STARRY-S** [rancher/rancher#46559]
**Rancher Server**
- Rancher version: `v2.9.0`

**Describe the bug**

Rancher agent image of 2.9.0 does not have windows `os.version` parameter in manifest list which may affect the deployment of windows clusters.

- Manifest list of image `rancher/rancher-agent:v2.9.0`:

    The windows platform does not have `os.version` parameter.

    ```console
    $ skopeo inspect docker://docker.io/rancher/rancher-agent:v2.9.0 --raw
    {
      "schemaVersion": 2,
      "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
      "manifests": [
        {
          "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
          "digest": "sha256:eb55523a6d7f09d027be43723853103a4046031981498f6185a75798cbb9d6a9",
          "size": 2211,
          "platform": {
            "architecture": "amd64",
            "os": "linux"
          }
        },
        {
          "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
          "digest": "sha256:c4e876e481a17ac30bb4b81a80fae5353e5da924fdce37d6a054fa3cd3ec8556",
          "size": 2210,
          "platform": {
            "architecture": "arm64",
            "os": "linux"
          }
        },
        {
          "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
          "digest": "sha256:7d0e2e627e9cea62b9f7cd847dea4606d5d80aa2f43cb6d5a2ae07e710664488",
          "size": 4301,
          "platform": {
            "architecture": "amd64",
            "os": "windows"
          }
        },
        {
          "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
          "digest": "sha256:10957d6dcb2d179d8b60caf496b8d5059fdb1dab9696510fe7644d83c6015e95",
          "size": 4301,
          "platform": {
            "architecture": "amd64",
            "os": "windows"
          }
        }
      ]
    }
    ```

- Manifest list of `rancher-agent:v2.8.5` has the `os.version` parameter in windows platform:

    ```console
    $ skopeo inspect docker://docker.io/rancher/rancher-agent:v2.8.5 --raw
    {
      "schemaVersion": 2,
      "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
      "manifests": [
        {
          "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
          "size": 1793,
          "digest": "sha256:3f4c1e8923f13eea8d31382b672f846133aca380d957b52011e91b2771f77a7a",
          "platform": {
            "architecture": "amd64",
            "os": "linux"
          }
        },
        {
          "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
          "size": 1793,
          "digest": "sha256:4ce46693d1115375ac9cf849a6a780bb542e9064798e0ff3c5a7efd0ee0118ad",
          "platform": {
            "architecture": "arm64",
            "os": "linux"
          }
        },
        {
          "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
          "size": 4300,
          "digest": "sha256:9b1ca0bf6f13128cd5c8e1ad3c92c84f698bf9b4e148c7d11a4bc55deafca029",
          "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.17763.5936"
          }
        },
        {
          "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
          "size": 4300,
          "digest": "sha256:bfba56e24ca3a42a17fad3a4a987638fc5a4292b0e1315af36bcc66b00bd5389",
          "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.20348.2527"
          }
        }
      ]
    }
    ```

Related code position of 2.9 GHA workflow: https://github.com/rancher/rancher/blob/v2.9.0/.github/workflows/push-release.yml#L383C6-L395C65

According to [the docs of docker-buildx](https://docs.docker.com/reference/cli/docker/buildx/imagetools/create/#file):

> In order to define annotations or additional platform properties like `os.version` and `os.features` you need to add them in the OCI descriptor object encoded in JSON.

**Comment by HarrisonWAffel** [comment]
This is a similar issue to https://github.com/rancher/system-agent-installer-rke2/pull/59. In short, `buildx` does not properly set the `os.version` value for Windows nodes, so the creation of the manifest needs to be done in two steps.

**Comment by STARRY-S** [comment]
Verified fixed in `v2.9.1`.

```console
$ skopeo inspect docker://docker.io/rancher/rancher-agent:v2.9.1-ent
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
  "manifests": [
    {
      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
      "digest": "sha256:e82701968c284fa7859fa1dfa6a5551184107ee07eb837754b896dd44302aee9",
      "size": 4301,
      "platform": {
        "architecture": "amd64",
        "os": "windows",
        "os.version": "10.0.17763.6189"
      }
    },
    {
      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
      "digest": "sha256:4428a078803b0aedbd5ba960ead9d098c2e773a1f1e87f0aa91fcfd44d2d5f8e",
      "size": 4301,
      "platform": {
        "architecture": "amd64",
        "os": "windows",
        "os.version": "10.0.20348.2582"
      }
    },
    {
      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
      "digest": "sha256:dcab9372b38053b7e51e5fc3685fe56d651bc6d6f28a4c13846d2544a2d41e72",
      "size": 2211,
      "platform": {
        "architecture": "amd64",
        "os": "linux"
      }
    },
    {
      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
      "digest": "sha256:6e60cfdc542a55977e99cbf151a224104198778125f334160d5e1e45b31c2b6c",
      "size": 2210,
      "platform": {
        "architecture": "arm64",
        "os": "linux"
      }
    }
  ]
}
```
